packages/gcc10.rb

```diff
@@ -82,7 +86,10 @@ def self.preinstallcheck
              installed_gccver.to_s == 'bash:' ||
              installed_gccver.to_s == @gcc_version.to_s ||
              installed_gccver.partition('.')[0].to_s == @gcc_version.partition('.')[0].to_s
-        abort "GCC version #{installed_gccver} is already installed. Please run: crew remove gcc#{installed_gccver} && crew install #{self.to_s.downcase}".lightgreen
+        $stderr.puts "GCC version #{installed_gccver} is currently installed.".lightred
+        $stderr.puts "To use #{self.to_s.downcase} please run:".lightgreen
+        $stderr.puts "crew remove gcc#{installed_gccver} && crew install #{self.to_s.downcase}".lightgreen
+        exit 1
```

Comment:
This is still too late in the process.  The warning should happen before the download.

Reply:
Is it not happening before the process?

Inside `crew` this is what is happening:
```
def install
  if !@pkg.in_upgrade && @device[:installed_packages].any? { |pkg| pkg[:name] == @pkg.name }
    puts "Package #{@pkg.name} already installed, skipping...".lightgreen
    return
  end

  unless @pkg.is_fake?
    # perform pre_install_check process to see if package should even
    # be downloaded.
    pre_install_check CREW_DEST_DIR
  end

  unless @pkg.is_fake?
    meta = download
    target_dir = unpack meta
    if meta[:source] == true
```

`meta = download` is where downloads happen, right? I put `pre_install_check` before that point.

Reply:
You could do the same exact thing in preflight without needing to add the exit 1.

Reply:
The ` $stderr.puts + exit 1` is there instead of using `abort` which equals `warn + exit 1`  as I understand it. I wanted the multiline output.

(Also rejiggering this to use preflight...)

Reply:
done with moving preflight around...